### PR TITLE
Dashboard integration

### PR DESCRIPTION
**Pull request by danielabreo**
Note: Testing for `memberGet` was not possible because we are unable to simulate the `request.user` object. Testing for `organizationMemberGet` was not possible because it uses firestore references which are not supported by our testing framework.

Adds 2 API's:
`/data/member`: returns the database ID of the currently logged in member.
`/data/member/organization/:id` returns the database ID of the organization assigned to the logged in member.

### CODE REVIEW

File: public/js/dashboardCategories.js [hunk 1]
```diff
@@ -3,22 +3,17 @@
...
-    },
-    {
-      ID: 'house-supplies',
-      name: 'Household Supplies',
-    },
-  ];
+async function getAllCategories() {
+  let responseData = await fetch('/data/categories');
+  let responseDataJSON = await responseData.json();
+  let allCategories = responseDataJSON.map(
```
Comment (SabrinaSnider) [inline public/js/dashboardCategories.js:9]: Nit: Can these be const?

File: routes/data.js [hunk 2]
```diff
@@ -114,3 +114,47 @@ exports.categoriesGet = async function (req, res) {
   });
   res.send(categories);
 };
+
+exports.member = async function (req, res) {
```
Comment (SabrinaSnider) [inline routes/data.js:118]: Nit: Change to `memberGet` to match the naming of the other route functions.

File: routes/data.js [hunk 3]
```diff
@@ -115,6 +115,48 @@ exports.categoriesGet = async function (req, res) {
   res.send(categories);
 };
 
+exports.memberGet = async function (req, res) {
```
Comment (brucezheng) [inline routes/data.js:118]: ?Could we name functions ```{verb}{noun}``` e.g. ```getMember```?

Comment (danielabreo) [reply routes/data.js]: Only changed the functions added in this PR to prevent merge conflicts / breaking functionality.

File: routes/data.js [hunk 4]
```diff
@@ -203,6 +203,48 @@ exports.organizationMemberGet = async function (req, res) {
...
+        .doc()
+        .set(userData)
+        .then(
+          memberSnapshot.get().then(function (doc) {
+            res.send(doc.docs[0].id);
+          })
+        );
+    }
+  });
+};
```
Comment (ccyang314) [inline routes/data.js:231]: nit: a general notes on the current backend logics. I think we need to add some inline documentations to track details on the implementation. As we are adding more code, we need to make sure the codebase has high readability and maintainability. 

In this block, I think the if/else blocks in line 216-230 might need some documentation on explaining its decision making logic.

**Pull request merged**